src/Microsoft.TemplateEngine.Edge/Settings/TemplateCache.cs

```diff
@@ -224,11 +249,17 @@ private void ScanForComponents(IMountPoint mountPoint, string templateDir)
             }
         }
 
-        public List<TemplateInfo> LoadTemplateCacheForLocale(string locale)
+        public List<TemplateInfo> LoadTemplateCacheForLocale(string locale, out string cacheVersion)
```

Comment:
Since this seems to ripple out a bit, hopefully we can internalize the version checks rather than having them passed in